File: SDKs/amazonaws.com/organizations/2016-11-28/python/src/sdk/models/operations/listchildren.py
from dataclasses import dataclass, field
from typing import Any,Enum,List,Optional
from sdk.models import shared


@dataclass
class ListChildrenQueryParams:
    max_results: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'MaxResults', 'style': 'form', 'explode': True }})
    next_token: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'NextToken', 'style': 'form', 'explode': True }})
    
class ListChildrenXAmzTargetEnum(str, Enum):
    AWS_ORGANIZATIONS_V20161128_LIST_CHILDREN = "AWSOrganizationsV20161128.ListChildren"


@dataclass
class ListChildrenHeaders:
    x_amz_algorithm: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Algorithm' }})
    x_amz_content_sha256: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Content-Sha256' }})
    x_amz_credential: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Credential' }})
    x_amz_date: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Date' }})
    x_amz_security_token: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Security-Token' }})
    x_amz_signature: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Signature' }})
    x_amz_signed_headers: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-SignedHeaders' }})
    x_amz_target: ListChildrenXAmzTargetEnum = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Target' }})
    

@dataclass
class ListChildrenRequest:
    query_params: ListChildrenQueryParams = field(default=None)
    headers: ListChildrenHeaders = field(default=None)
    request: shared.ListChildrenRequest = field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    

@dataclass
class ListChildrenResponse:
    aws_organizations_not_in_use_exception: Optional[Any] = field(default=None)
    access_denied_exception: Optional[Any] = field(default=None)
    content_type: str = field(default=None)
    invalid_input_exception: Optional[Any] = field(default=None)
    list_children_response: Optional[shared.ListChildrenResponse] = field(default=None)
    parent_not_found_exception: Optional[Any] = field(default=None)
    service_exception: Optional[Any] = field(default=None)
    status_code: int = field(default=None)
    too_many_requests_exception: Optional[Any] = field(default=None)
    

File: SDKs/amazonaws.com/organizations/2016-11-28/go/pkg/models/operations/listchildren.go
package operations

import (
	"openapi/pkg/models/shared"
)

type ListChildrenQueryParams struct {
	MaxResults *string `queryParam:"style=form,explode=true,name=MaxResults"`
	NextToken  *string `queryParam:"style=form,explode=true,name=NextToken"`
}

type ListChildrenXAmzTargetEnum string

const (
	ListChildrenXAmzTargetEnumAwsOrganizationsV20161128ListChildren ListChildrenXAmzTargetEnum = "AWSOrganizationsV20161128.ListChildren"
)

type ListChildrenHeaders struct {
	XAmzAlgorithm     *string                    `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string                    `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string                    `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string                    `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string                    `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string                    `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string                    `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
	XAmzTarget        ListChildrenXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
}

type ListChildrenRequest struct {
	QueryParams ListChildrenQueryParams
	Headers     ListChildrenHeaders
	Request     shared.ListChildrenRequest `request:"mediaType=application/json"`
}

type ListChildrenResponse struct {
	AwsOrganizationsNotInUseException *interface{}
	AccessDeniedException             *interface{}
	ContentType                       string
	InvalidInputException             *interface{}
	ListChildrenResponse              *shared.ListChildrenResponse
	ParentNotFoundException           *interface{}
	ServiceException                  *interface{}
	StatusCode                        int64
	TooManyRequestsException          *interface{}
}
